### DISCUSSION TITLE
Date selector is clunky to use.

### DISCUSSION THREAD

**Discussion by CodeMonkeyOne** [trakt/showly#837] [Ideas / Feature Requests]
Issue Description: When marking an item as watched on custom date, the date selector allows YEAR to choose and directly moves to DATE of current month is the selected year. This adds extra steps of changing to the correct month using the provided arrow buttons. This process is tiring when I'm adding dates for multiple items at once.

Screenshots:

![Screenshot_20240814_181338_Showly](https://github.com/user-attachments/assets/0b0cd418-346e-4326-b74e-755e44daf917)

>>

![Screenshot_20240814_181343_Showly](https://github.com/user-attachments/assets/abfbcb42-f15e-437a-975d-69c3fb800ccc)

**Expected behavior**
A clear heirachical date selection from DECADE (eg. 2010 - 2020) > YEAR > MONTH > DATE

Additionally, if the release date of the selected item (Movie or Tv show) can be displayed along with date selector, it would be helpful when I'm trying to mark it as the date after release ( which is a common scenario).

Thanks for the wonderful app. Looking forward to new updates on Play Store.

**Comment by michaldrabik** [comment]
Hello and thank you. To be honest I am using latest standard calendar component from Google Android Material so I don't have much control over this. 

Not sure if you noticed but you can press the pencil icon when selecting date and it will bring alternative input option. Which might be even faster for You.

**Comment by CodeMonkeyOne** [comment]
Welp. That component definitely sucks then. I did try the pencil to type in which is a disaster of its own. It replaces deleted values with hard letters (mm/dd/yyyy). I'm attaching a screen grab of that issue belowm

https://github.com/user-attachments/assets/74bd12aa-0cc5-40f8-a97b-289ad888a6e3

Nonetheless, if we can see the release date over the selector, or have a direct option to mark watched on release date, that could improve the experience a lot.

Thanks again!

**Comment by michaldrabik** [comment]
Adding 3rd option for release date would probably be most sane. Will be considering it.

**Comment by CodeMonkeyOne** [comment]
Thank you.
Just piggybacking on this thread, when might we'll have the latest github version on play store? I'm using this without premium for the date selector.

**Reply by michaldrabik** [reply]
@CodeMonkeyOne Latest PlayStore version is 3.38.0 available for everyone. It contains date selector of course.
